cmd/cli/install.go

```diff
@@ -189,7 +198,7 @@ func (i *installCmd) loadOSMChart() error {
 	return nil
 }
 
-func (i *installCmd) resolveValues() (map[string]interface{}, error) {
+func (i *installCmd) resolveValues(parsedArgs []string) (map[string]interface{}, error) {
 	finalValues := map[string]interface{}{}
 	valuesConfig := []string{
 		fmt.Sprintf("OpenServiceMesh.image.registry=%s", i.containerRegistry),
```

Comment:
I think if `valuesConfig` was `i.setOptions` + all of the existing `fmt.Sprintf` calls for the explicit flags (in that order), then you wouldn't have to manually override the value from `--set` with the explicit flag since Helm would do that based on the order (not 100% sure about that). That would eliminate the need for `overrideLegacyArgs()` where the variable/chart value association is duplicated, and maybe even `parseSetOptions()`.

Reply:
Same here.. `validateOptions` operates on the instance variables.. whereas `valuesConfig` array has formatted strings with both key and value combines... validating such a format makes it tricky... let me know